### PR TITLE
[MM-60603] Don't follow threads when marking them as read on focus

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
I made some changes to mark threads as read on focus, which revealed [this change](https://github.com/mattermost/mattermost/pull/17800) where we create a thread membership upon marking as read if it doesn't exist. This was causing some weird behaviour where we would auto-follow threads that the user is reading on focus, even if the user wasn't following the thread.

It makes more sense to me to just check for the thread membership before we try to mark the thread as read, and throw a Not Found if we don't have a membership. I also made a change to check if we are following the thread before marking as read on focus, since we won't have a thread membership otherwise.

One thing I do find odd is that having a thread membership basically implies that we are following the thread - so I'm not sure why we have the flag on there (perhaps done for historical reasons?). If there is no reason for that this would be something worth refactoring at some point.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-60603

```release-note
Don't follow threads when marking them as read on focus
```

### CODE REVIEW

**Review by harshilsharma63 — APPROVED** [review]
I tested this locally and it works perfectly!

LGTM!

**Review by yasserfaraazkhan — APPROVED** [review]

**Review by hmhealey — CHANGES_REQUESTED** [review]
This is intentionally removing the original fix for https://mattermost.atlassian.net/browse/MM-36430, so we still need to fix that some other way. If you try the repro steps for that bug, you'll see the failed request in the console and the thread will be marked as Followed on the web app but refreshing will make that go away.

We can either:
1. Keep the existing behaviour if the user marks the thread as specifically unread but not automatically follow the thread if marking as read. This follows the spirit of the original fix, but I'm not of the API still having a likely unexpected side effect.
2. Keep the same UX as above, but instead of having the server follow the thread, make the web app make two separate API calls: one to follow the thread and then a second to mark it as unread. That seems like a mix of the nicest UX and not making the API have unexpected side effects
3. Remove the "Mark as Unread" option from the post menu for threads that you're not following.

I think I'm favouring fix 2 at the moment, but I'm only 1/5 towards that

**Comment by devinbinnie** [comment]
> This is intentionally removing the original fix for https://mattermost.atlassian.net/browse/MM-36430, so we still need to fix that some other way. If you try the repro steps for that bug, you'll see the failed request in the console and the thread will be marked as Followed on the web app but refreshing will make that go away.
> 
> We can either:
> 
> 1. Keep the existing behaviour if the user marks the thread as specifically unread but not automatically follow the thread if marking as read. This follows the spirit of the original fix, but I'm not of the API still having a likely unexpected side effect.
> 2. Keep the same UX as above, but instead of having the server follow the thread, make the web app make two separate API calls: one to follow the thread and then a second to mark it as unread. That seems like a mix of the nicest UX and not making the API have unexpected side effects
> 3. Remove the "Mark as Unread" option from the post menu for threads that you're not following.
> 
> I think I'm favouring fix 2 at the moment, but I'm only 1/5 towards that

I think given those options I'll take number 2 as well, since it removes the unnecessary side effect and doesn't change other existing intended behaviour. Thanks for catching that :)

**Comment by devinbinnie** [comment]
@yasserfaraazkhan Given Harrison's requested changes can you do a super quick re-test?

File: server/channels/api4/user_test.go
Comment (hmhealey) [inline server/channels/api4/user_test.go:1]: Could the changes to this file be reverted now since we re-added the following behaviour on the server?

**Review by hmhealey — APPROVED** [review]
Looks good outside of one last question about the API unit test

File: server/channels/api4/user_test.go
Comment (devinbinnie) [reply server/channels/api4/user_test.go:1]: Don't think so - again the API call to `set_unread` is different than the one that was causing the issue.

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**